hosting/Estranged.Lfs.Hosting.AspNet/Startup.cs

```diff
@@ -23,7 +23,19 @@ public void ConfigureServices(IServiceCollection services)
             });
 
             services.AddSingleton<IAmazonS3, AmazonS3Client>();
-            services.AddLfsS3Adapter(new S3BlobAdapterConfig{Bucket = "estranged-lfs-test"}, new AmazonS3Client());
+            const string LfsBucket = "estranged-lfs-test";
+            const string S3AccessKeyId = "";
+            const string S3AccessKeySecret = "";
+            const string S3Region = "";
+            const string S3ServiceURL = "";
+            if (!string.IsNullOrWhiteSpace(S3ServiceURL) && !string.IsNullOrWhiteSpace(S3Region) && !string.IsNullOrWhiteSpace(S3AccessKeyId) && !string.IsNullOrWhiteSpace(S3AccessKeySecret))
+            {
+                services.AddLfsS3Adapter(new S3BlobAdapterConfig { Bucket = LfsBucket }, new AmazonS3Client(S3AccessKeyId, S3AccessKeySecret, new AmazonS3Config { ServiceURL = S3ServiceURL, AuthenticationRegion = S3Region, SignatureVersion = "V4" }));
+            }
+            else
+            {
+                services.AddLfsS3Adapter(new S3BlobAdapterConfig { Bucket = LfsBucket }, new AmazonS3Client());
+            }

```

Comment:
The point of these being [packages on NuGet](https://www.nuget.org/packages?q=estranged.lfs) is so that this kind of granular customisation can happen outside of this repository (e.g. in your own application). I think this is too application specific to be useful in the sample application.

Reply:
You are just using the default values, which only restrict to AWS S3.
It opens the configuration with optional parameters without breaking the existing AWS one.
With this solution, anyone could just clone the repo and is ready to go (required only to change the parameters to fit its cloud provider).